tests/basic_cxx_only/cpp/main.cpp

```diff
@@ -18,9 +17,19 @@ get_cpp_number()
   return hidden_num;
 }
 
-TEST_CASE("Clean cxx allows basic interaction between C++ and Rust")
+class CxxTest : public QObject
 {
-  CHECK(get_numbers_sum() == 102);
-  hidden_num = 200;
-  CHECK(get_numbers_sum() == 202);
-}
+  Q_OBJECT
+
+private Q_SLOTS:
+  // Clean cxx allows basic interaction between C++ and Rust
+  void test_cxx_interaction()
+  {
+    QCOMPARE(get_numbers_sum(), 102);
+    hidden_num = 200;
```

Comment:
So this is an old test that i've just ported. What it does is ....
* Calls `get_numbers_sum()` (Rust)
  * Which calls `get_cpp_number()` (C++) and returns + 2
    * `get_cpp_number()` returns `hidden_num`
  * So we should get 102 as `hidden_num` starts as 100
* Changes `hidden_num` to 200
* Calls `get_numbers_sum()` (Rust)
  * Which calls `get_cpp_number()` (C++) and returns + 2
    * `get_cpp_number()` returns `hidden_num`
  * So we should get 202 as `hidden_num` is now 200

It looks like `hidden_sum` is unused, but it's checking that the basic CXX bridge works as we expect back and forth across the bridge.

Eventually I want to merge this small test with basic_cxx_qt, as it doesn't warrant it's own project folder anymore.

But the point of this PR was to port the tests to QTest, so that change can be in later changes.

Reply:
okay